src/scores/probability/crps_impl.py

```diff
@@ -735,3 +737,87 @@ def crps_step_threshold_weight(
         weight = 1 - weight
 
     return weight
+
+
+def crps_for_ensemble(
+    fcst: XarrayLike,
```

Comment:
`XarrayLike` means that the input can be a DataArray or a DataSet.

You need to either add a test that checks that it works when then inputs are DataSets or update the typehint to just be xr.DataArray.

Reply:
I think it is easiest to assume data inputs are xr.DataArray, even though this is not the most general implementation possible. I've updated typehints accordingly.